plugin/session_view.py

```diff
@@ -80,8 +80,10 @@ def on_before_remove(self) -> None:
             self.session.unregister_session_view_async(self)
         self.session.config.erase_view_status(self.view)
         for severity in reversed(range(1, len(DIAGNOSTIC_SEVERITY) + 1)):
-            self.view.erase_regions(self.diagnostics_key(severity, False))
-            self.view.erase_regions(self.diagnostics_key(severity, True))
+            self.view.erase_regions("{}_icon".format(self.diagnostics_key(severity, False)))
+            self.view.erase_regions("{}_underline".format(self.diagnostics_key(severity, False)))
+            self.view.erase_regions("{}_icon".format(self.diagnostics_key(severity, True)))
+            self.view.erase_regions("{}_underline".format(self.diagnostics_key(severity, True)))
```

Comment:
Hmm,
I kind of liked the previous code more (mostly because it was less code)
In the prev code, things were done twice, with this change things needs to be handled x4. 

Is there a way to fix it differently with out introducing new regions for `_icon` and `_underline`?

Reply:
I can't really think of another way how it could be solved in practice. The problem is basically that ST behaves inconsistent in regard to which of the gutter icons and which of the inline decorations it will render (on top) when there are multiple regions overlapping. If ST would make this behavior consistent, or give the plugin more control to for example explicitly set the priority for regions, then this workaround wouldn't be required.

In theory another approach could be to compare the different diagnostics for a file, and only draw the single one with the highest severity, if there are multiple diagnostics with different severities and the same range. But I imagine that this would be quite computationally expensive. And the ranges could also be partially overlapping, then you would have to calculate the overlapping range and potentially generate new ranges by subtracting them, etc., so I think this would be infeasable (not even considering diagnostics from multiple sessions...).

If you have another idea, suggestions are welcome, of course.

Reply:
I was also thinking about sorting the diagnostics by severity and only draw the first with the highest severity,
but you have a good point with:
> And the ranges could also be partially overlapping, then you would have to calculate the overlapping range and potentially generate new ranges by subtracting them, etc.,